python/lsst/decam/hits/selectImages.py

```
@@ -0,0 +1,101 @@
+
+    def runDataRef(self, dataRef, coordList, makeDataRefList=True, selectDataList=[]):
+        """Select images in the selectDataList that overlap the patch
+
+        We use the "convexHull" function in the geom package to define
+        polygons on the celestial sphere, and test the polygon of the
+        patch for overlap with the polygon of the image.
+
+        We use "convexHull" instead of generating a SphericalConvexPolygon
+        directly because the standard for the inputs to SphericalConvexPolygon
+        are pretty high and we don't want to be responsible for reaching them.
+        If "convexHull" is found to be too slow, we can revise this.
+
+        @param dataRef: Data reference for coadd/tempExp (with tract, patch)
+        @param coordList: List of Coord specifying boundary of patch
+        @param makeDataRefList: Construct a list of data references?
+        @param selectDataList: List of SelectStruct, to consider for selection
+        """
+        from lsst.geom import convexHull
+        psf_sizes = []
+        dataRefList = []
+        exposureInfoList = []
+
+        patchVertices = [coord.getVector() for coord in coordList]
+        patchPoly = convexHull(patchVertices)
+
+        for data in selectDataList:
+            dataRef = data.dataRef
+            imageWcs = data.wcs
+            cal = dataRef.get('calexp', immediate=True)
+            psf_size = cal.getPsf().computeShape().getDeterminantRadius()
+            nx, ny = data.dims
+
+            imageBox = afwGeom.Box2D(afwGeom.Point2D(0, 0), afwGeom.Extent2D(nx, ny))
+            try:
+                imageCorners = [imageWcs.pixelToSky(pix) for pix in imageBox.getCorners()]
+            except (pexExceptions.DomainError, pexExceptions.RuntimeError) as e:
+                # Protecting ourselves from awful Wcs solutions in input images
+                self.log.debug("WCS error in testing calexp %s (%s): deselecting", dataRef.dataId, e)
+                continue
```

Are you sure it's safe to silently ignore corrupted data? I'm especially worried about what will happen if this Task returns empty lists.

---

Empirically, it just returns

```makeCoaddTempExp WARN: No exposures to coadd for patch DataId(initialdata={'filter': 'g', 'patch': '0,0', 'tract': 0}, tag=set([]))```

Which I think is fine.

---

You mean, the rest of the pipeline just does nothing until it finishes successfully?

---

`selectImages` is passing `dataRef`s up the `makeCoaddTempExp`.  If it returns empty lists because the data is bad, `makeCoaddTempExp` warns that there is no data for that patch and continues along.  (This can also happen if there is in fact no data there.)

---

I would have thought that "no data" should be something alarming, but I think you're saying this is standard policy. Fine, objection withdrawn.